internal/migrations/migrate_clusterlogforwarder.go

```diff
@@ -0,0 +1,59 @@
+package migrations
+
+import (
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+)
+
+func MigrateClusterLogForwarderSpec(spec logging.ClusterLogForwarderSpec) logging.ClusterLogForwarderSpec {
+	spec.Outputs = MigrateDefaultOutput(spec)
+	return spec
+}
+
+// MigrateDefaultOutput adds the 'default' output spec to the list of outputs if it is not defined or
+// selectively replaces it if it is.  It will apply OutputDefaults unless they are already defined.
+func MigrateDefaultOutput(spec logging.ClusterLogForwarderSpec) []logging.OutputSpec {
+
+	refDefault := false
+	for _, p := range spec.Pipelines {
+		for _, ref := range p.OutputRefs {
+			if ref == logging.OutputNameDefault {
+				refDefault = true
+				break
+			}
+		}
+	}
```

Comment:
Not a blocker for merging but FYI there's some helper code for stuff like this:
```suggestion
_,  refDefault := logging.v1.NewRoutes(spec.Pipelines).ByOutput[logging.OutpuNameDefault]
```